src/datumaro/experimental/fields/annotations.py

```
@@ -424,3 +429,53 @@ def from_polars(self, name: str, row_index: int, df: pl.DataFrame, target_type:
         """Reconstruct an ellipse tensor from Polars data."""
         polars_data = df[name][row_index]
         return from_polars_data(polars_data, target_type)  # type: ignore
+
+
+@dataclass(frozen=True)
+class CaptionField(Field):
+    """
+    Represents a text caption field.
+
+    Stores either a single caption string or a list of caption strings when
+    is_list=True. Useful for tasks like image captioning, multi-caption datasets,
+    or storing alternative textual descriptions.
+
+    Attributes:
+        semantic: Semantic tags describing the caption's purpose
+        is_list: Whether this field stores multiple captions (list[str])
+    """
+
+    semantic: Semantic
+    is_list: bool = False
+
+    def to_polars_schema(self, name: str) -> dict[str, pl.DataType]:
+        """Generate schema for caption column (string or list of strings)."""
+        dtype = pl.List(pl.Utf8()) if self.is_list else pl.Utf8()
+        return {name: dtype}
+
+    def to_polars(self, name: str, value: Any) -> dict[str, pl.Series]:
+        """Convert caption value(s) to Polars series."""
+        dtype = pl.List(pl.Utf8()) if self.is_list else pl.Utf8()
+        return {name: pl.Series(name, [value], dtype=dtype)}
+
+    def from_polars(self, name: str, row_index: int, df: pl.DataFrame, target_type: type[T]) -> T:
+        """Reconstruct caption value(s) from Polars data."""
+        data = df[name][row_index]
+        if self.is_list and target_type is list:
+            return list(data) if data is not None else None  # type: ignore[return-value]
```

Is `from_polars` expected to ever return `None`? If yes, I guess the function return signature should be `-> T | None`, else you can raise an exception if data is `None`.